src/main/scala/io/iohk/ethereum/vm/ProgramContext.scala

```diff
@@ -1,13 +1,51 @@
 package io.iohk.ethereum.vm
 
-import akka.util.ByteString
+import io.iohk.ethereum.domain._
 
+
+object ProgramContext {
+  def apply(stx: SignedTransaction, blockHeader: BlockHeader, world: WorldStateProxy): ProgramContext = {
+    import stx.tx
+
+    val senderAddress = stx.recoveredSenderAddress.get // FIXME: get, it should be validated but...
```

Comment:
2 questions that have appeared while implementing this:
- What is the use case for an invalid `SignedTransaction`?
- Do we still need the `Transaction` / `SignedTransaction` separation?